tests/components/androidtv/test_media_player.py

```diff
@@ -1167,9 +1167,10 @@ async def test_exception(hass: HomeAssistant) -> None:
         # When an unforeseen exception occurs, we close the ADB connection and raise the exception
         with patchers.PATCH_ANDROIDTV_UPDATE_EXCEPTION, pytest.raises(Exception):
             await async_update_entity(hass, entity_id)
-            state = hass.states.get(entity_id)
-            assert state is not None
-            assert state.state == STATE_UNAVAILABLE
+
+        state = hass.states.get(entity_id)
+        assert state is not None
+        assert state.state == STATE_UNAVAILABLE
```

Comment:
Hello @JeffLIrion, @ollo69, can you look at this test? It doesn't seem to work at all. The `ZeroDivisionError` from `patchers.PATCH_ANDROIDTV_UPDATE_EXCEPTION` is not raised, the `pytest.raises` captured the failing `assert state.state == STATE_UNAVAILABLE` because the `pytest.raises(Exception)` is much too broad and the asserts were inside the `pytest.raises` context. The state is not changing to unavailable, it is still `'off'`.

Reply:
The exception is not raised because catch by entity class at line 952. I create PR #114133 to fix behavior in decorator, not really sure that is the right approach.